src/main/java/org/sagebionetworks/web/client/widget/ReactComponentV2.java

```diff
@@ -92,11 +92,40 @@ private void createRoot() {
     }
   }
 
+  /**
+   * Asynchronously (in the task queue, via setTimeout) unmounts the root and sets it to null.
+   */
   private void destroyRoot() {
```

Comment:
Just as in ReactComponent (v1), move the asynchronous logic inside of `destroyRoot`